core/mm3Types.js

```diff
@@ -212,12 +361,83 @@ CLASS({
 });
 
 
+
+
 CLASS({
-  name:  'IntProperty',
+  name:  'NumericProperty_',
   extends: 'Property',
 
+  help:  'Base model for a property of any numeric type.',
+
+  messages: [
+    {
+      name: 'errorBelowMinimum',
+      value: 'The value must be at least $min$.',
+      placeholders: [ { name: 'min' } ]
+    },
+    {
+      name: 'errorAboveMaximum',
+      value: 'The value can be at most $max$.',
+      placeholders: [ { name: 'max' } ]
+    }
+  ],
+
+  properties: [
+    {
+      name: 'minValue',
+      label: 'Minimum Value',
+      required: false,
+      help: 'The minimum value this property accepts.',
+      defaultValue: ''
+    },
+    {
+      name: 'maxValue',
+      label: 'Maximum Value',
+      required: false,
+      help: 'The maximum value this property accepts.',
+      defaultValue: ''
+    },
+    {
+      name: 'compareProperty',
+      labels: ['javascript'],
+      defaultValue: function(o1, o2) { return o1 === o2 ? 0 : o1 > o2 ? 1 : -1; },
+    },
+    {
+      name: 'validate',
+      lazyFactory: function() {
+        var prop = this; // this == the property
+        var ret = constantFn('');
+
+        var min = prop.minValue;
+        if ( typeof min !== 'string' || min ) { // ignore '' (falsey string)
```

Comment:
You could change your || to min == "", which would make the code self explanatory and then remove the comment.